src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -70,19 +70,31 @@ trait Contexts { self: Analyzer =>
   private lazy val allImportInfos =
     mutable.Map[CompilationUnit, List[ImportInfo]]() withDefaultValue Nil
 
+  private def checkDeprecatedImport(qual: Tree, selector: ImportSelector): List[String] = {
+    var messages: List[String] = Nil
+    if (qual.symbol.isDeprecated)
+      messages = s"import from deprecated ${qual.symbol}${qual.symbol.deprecationMessage.map(": " + _).getOrElse("")}" :: messages
+    Iterator(selector.name, selector.name.toTypeName).map(qual.tpe.member(_)).collectFirst {
+      case m if m.isDeprecated => s"import of deprecated $m${m.deprecationMessage.map(": " + _).getOrElse("")}"
+    }.foreach(s => messages = s :: messages)
+    messages
+  }
   @tailrec private def warnUnusedImportInfos(infos: List[ImportInfo]): Unit =
     infos match {
       case info :: rest =>
         val used = allUsedSelectors(info)
         @tailrec def loop(selectors: List[ImportSelector]): Unit =
           selectors match {
             case selector :: rest =>
-              if (!selector.isMask && !used(selector))
-                reporter.warning(info.posOf(selector), "Unused import")
+              if (!selector.isMask && !used(selector)) {
+                val more = if (selector.isSpecific) checkDeprecatedImport(info.qual, selector) else Nil
```

Comment:
Actually an `import foo.syntax._` that is deprecated because unused should show a message if `syntax` is deprecated. For example, `import scala.collection.Searching._` which was seen in a test.